genesis/process/shardGenesisBlockCreator.go

```diff
@@ -418,7 +420,22 @@ func createProcessorsForShardGenesisBlock(arg ArgsGenesisBlockCreator, enableEpo
 		return nil, err
 	}
 
-	argsNewVMFactory := shard.ArgVMContainerFactory{
+	messageSignVerifier, err := disabled.NewMessageSignVerifier(arg.BlockSignKeyGen)
+	if err != nil {
+		return nil, err
+	}
+
+	// VM container only needs to know the number of eligible nodes.
+	// At genesis, we don't need a full nodes coordinator, and this information
+	// is already available in initial nodes setup from genesis
+	liteNodesCoordinator := &shardingMocks.NodesCoordinatorMock{
```

Comment:
Please do not use the Mock here... someone might change this and look for effects just in the tests...

Reply:
use a disabled component here. not a mock.

Reply:
We also use in other parts of the code mocks in productive code.
Created a new small comp to be used